test/acceptance/application.test.ts

```diff
@@ -19,375 +20,82 @@ describe('Controllers', () => {
-        await client
-          .post('/todo')
-          .send({})
-          .expect(400);
-      });
-
-      it('returns 409 if the item already exists', async () => {
-        const todo = (await client.post('/todo').send(payload)).body;
-        await client
-          .post('/todo')
-          .send(todo)
-          .expect(409);
-      });
-    });
-
-    describe('findById', () => {
-      it('returns a todo if it exists', async () => {
-        const item = await client.post('/todo').send({
-          title: 'fake'
-        });
-        await client
-          .get(`/todo/${item.body.id}`)
-          .send()
-          .expect(200);
-      });
-
-      it('returns 404 if no match exists', async () => {
-        await client
-          .get(`/todo/9999999`)
-          .send()
-          .expect(404);
-      });
-    });
-
-    describe('find', () => {
-      it('finds everything', async () => {
-        await client
-          .get('/todo')
-          .send()
-          .expect(200);
-      });
-    });
-
-    describe('replace', () => {
-      it('successfully replaces existing items', async () => {
-        const item = await client.post('/todo').send({
-          title: 'another one'
-        });
-
-        await client
-          .put(`/todo/${item.body.id}`)
-          .send({
-            title: 'another two'
-          })
-          .expect(200);
-      });
-
-      it('returns 404 if no match exists', async () => {
-        await client
-          .put('/todo/999999')
-          .send({
-            title: 'another one'
-          })
-          .expect(404);
-      });
-    });
-
-    describe('update', () => {
-      it('successfully updates existing items', async () => {
-        const item = await client.post('/todo').send({
-          title: 'another one'
-        });
-
-        await client
-          .patch(`/todo/${item.body.id}`)
-          .send({
-            title: 'another two'
-          })
-          .expect(200);
-      });
-
-      it('returns 404 if no match exists', async () => {
-        await client
-          .patch('/todo/999999')
-          .send({
-            title: 'another one'
-          })
-          .expect(404);
-      });
-    });
-
-    describe('delete', () => {
-      it('successfully deletes matching item', async () => {
-        await client
-          .del('/todo/1')
-          .send()
-          .expect(200);
-      });
-
-      it('returns 404 if no match exists', async () => {
-        await client
-          .del('/todo/999999')
-          .send({
-            title: 'another one'
-          })
-          .expect(404);
-      });
-    });
+  const payload = {
+    title: 'do a thing',
+    desc: 'There are some things that need doing',
+    isComplete: false
+  };
+
+  it('creates a todo', async () => {
+    await client
+      .post('/todo')
+      .send(payload)
+      .expect(
+        200,
+        Object.assign({}, payload, {
+          id: 1
```

Comment:
I think it's a bad idea to hard-code id numbers in acceptance tests. If we change the persistence config later (e.g. use a different DB that uses UUID instead of numeric ids, or pre-populate database with some initial Todo instances), then we will have to rework all tests like this one.

Here is an assertion that may be better:

```js
const response = await client.post(...)
  // ...
  expect(200);

// all properties from the request were persisted
expect(response.body).to.containEql(payload);

// verify the record was stored
const repo = (await app.get('repositories.todo')) as TodoRepository;
const todoCreated = await TodoRepository.findById(response.body.id);
expect(todoCreated).to.containEql(payload);
```

Reply:
I thought I changed all of those; could be a bad merge.